samples/metadata-sample.c

```diff
@@ -5,35 +5,24 @@
 #include <assert.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include <pthread.h>
 
-#define FDB_API_VERSION 620
-#include <foundationdb/fdb_c.h>
 
 #include "util.h"
 #include "metadata.h"
-#include "metadata-foundationdb.h"
-
-void *run_net(void *_unused)
-{
-    chk(fdb_run_network());
-
-    return NULL;
-}
 
 int main()
 {
-    fdb_select_api_version(FDB_API_VERSION);
+    int ret = 0;
 
-    chk(fdb_setup_network());
+    if ((ret = use_metadata_storage_backend("foundationdb")) != 0) {
+        printf("Couldn't load foundationdb backend!\n");
+        exit(-1);
+    }
 
-    pthread_t net_thread;
-    assert(0 == pthread_create(&net_thread, NULL, run_net, NULL));
+    Metadata *metadata = new_metadata();
```

Comment:
that's already better however I thought about doing after `metadata_backend_initialize();` two lines further below

Reply:
you're right. I've now improved the error handling. Please take a look.